encoder/generator.go

```diff
@@ -456,7 +456,7 @@ func (g *generator) genInterface(name string, v cue.Value) {
 	// place in the body of the generated typescript interface. (Or nil, if
 	// there's no body to generate.)
 	for fields != nil && fields.Next() {
-		if fields.IsHidden() {
+		if fields.Selector().PkgPath() != "" {
```

Comment:
is this the new canonical way to tell if a field is hidden? It seems like this is checking something else

Reply:
I checked the code comments.
```
// IsHidden reports if a field is hidden from the data model.
//
// Deprecated: use i.Selector().PkgPath() != ""
```

Reply:
With the test it seems behaving the same, https://github.com/ying-jeanne/cue_test_cases/blob/d4c23b79200cc30d5a102b9ee90ad197e03da218/test2.go.

When fields are hidden, there pkgpath is _ instead of empty

Reply:
Gotcha, thanks for the thoroughness! i'll chalk this up to some kinda continued awkwardness in the CUE APIs around hidden fields